src/module.c

```diff
@@ -519,13 +528,12 @@ int SynUpdateCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   SynonymMap_UpdateRedisStr(sp->smap, argv + offset, argc - offset, id);
 
   if (initialScan) {
-    IndexSpec_ScanAndReindex(ctx, sp);
+    IndexSpec_ScanAndReindex(ctx, ref);
```

Comment:
Shouldn't it get a weak ref?

Reply:
internally in the creation of the scanner (and only if one is created) it takes a weak ref from this one